### PR TITLE
Docker enabled Cassandra Cluster

### PR DESCRIPTION
**Pull request by yaron2**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government) V
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken V
1. Use resourceGroup().location for resource locations Vv
1. Folder names for artifacts (nestedtemplates, scripts, DSC) 
1. Use literal values for apiVersion (no variables)  V
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values V
1. $schema and other uris use https V
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. V
1. Update the metadata.json with the current date V

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by yaron2** [comment]
Ready for review & merge.

**Comment by yaron2** [comment]
@bmoore-msft hey, can you take a look at this please? Thanks!

File: 101-cassandra-existing-vnet/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+  "variables": {
+    "publicIPAddressName": "loadBalancerPublicIP",
+    "loadBalancerName": "cassandraLoadBalancer",
+    "loadBalancerFrontEndName": "loadBalancerFrontEnd",
+    "loadBalancerBackEndName": "loadBalancerBackEnd",
+    "loadBalancerProbeName": "loadBalancerHttpProbe",
+    "loadBalancerNatPoolName": "loadBalancerNatPool",
+    "namingInfix": "[toLower(parameters('clusterName'))]",
+    "nicName": "[concat(variables('namingInfix'), '-nic')]",
+    "_artifactsLocation": "https://raw.githubusercontent.com/yaron2/azure-quickstart-templates/cassandra-cluster/101-cassandra-existing-vnet/scripts",
```
Comment (bmoore-msft) [inline 101-cassandra-existing-vnet/azuredeploy.json]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc) for how to handle artifacts

File: 101-cassandra-existing-vnet/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    "_extensionScript": "cassandraSetup.sh",
+    "nestedTemplateLocation": "https://raw.githubusercontent.com/yaron2/azure-quickstart-templates/cassandra-cluster/101-cassandra-existing-vnet/nested/cassandra-extension-vmss.json",
```
Comment (bmoore-msft) [inline 101-cassandra-existing-vnet/azuredeploy.json]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc) for how to handle artifacts

File: 101-cassandra-existing-vnet/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+    "masterVM": "[concat('Microsoft.Compute/virtualMachineScaleSets/', variables('namingInfix'), '/virtualMachines/1/networkInterfaces/', variables('nicName'))]",
+    "_artifactsLocationSasToken": ""
```
Comment (bmoore-msft) [inline 101-cassandra-existing-vnet/azuredeploy.json]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc) for how to handle artifacts

File: 101-cassandra-existing-vnet/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Compute/virtualMachineScaleSets",
+      "name": "[variables('namingInfix')]",
+      "location": "[resourceGroup().location]",
+      "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline 101-cassandra-existing-vnet/azuredeploy.json]: use a non-preview apiVersion

File: 101-cassandra-existing-vnet/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+                  "ipConfigurations": [
+                    {
+                      "name": "ipconfig",
+                      "properties": {
+                        "subnet": {
+                          "id": "[resourceId(parameters('existingVirtualNetworkResourceGroup'), 'Microsoft.Network/virtualNetworks/subnets', parameters('existingVirtualNetworkName'), parameters('existingSubnetName'))]"
+                        },
+                        "loadBalancerBackendAddressPools": [
+                          {
+                            "id": "[concat('/subscriptions/', subscription().subscriptionId,'/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Network/loadBalancers/', variables('loadBalancerName'), '/backendAddressPools/', variables('loadBalancerBackEndName'))]"
```
Comment (bmoore-msft) [inline 101-cassandra-existing-vnet/azuredeploy.json]: simplify - just use resourceId and remove subscriptionId and resourceGroup().name - resourceId function will add those

File: 101-cassandra-existing-vnet/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+                          }
+                        ],
+                        "loadBalancerInboundNatPools": [
+                          {
+                            "id": "[concat('/subscriptions/', subscription().subscriptionId,'/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Network/loadBalancers/', variables('loadBalancerName'), '/inboundNatPools/', variables('loadBalancerNatPoolName'))]"
```
Comment (bmoore-msft) [inline 101-cassandra-existing-vnet/azuredeploy.json]: simplify - just use resourceId and remove subscriptionId and resourceGroup().name - resourceId function will add those

File: 101-cassandra-existing-vnet/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+          {
+            "name": "[variables('loadBalancerBackendName')]"
+          }
+        ],
+        "loadBalancingRules": [
+          {
+            "name": "roundRobinLBRule",
+            "properties": {
+              "frontendIPConfiguration": {
+                "id": "[concat(resourceId('Microsoft.Network/loadBalancers', variables('loadBalancerName')), '/frontendIPConfigurations/', variables('loadBalancerFrontEndName'))]"
```
Comment (bmoore-msft) [inline 101-cassandra-existing-vnet/azuredeploy.json]: simplify - just use resourceId

File: 101-cassandra-existing-vnet/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+              },
+              "backendAddressPool": {
+                "id": "[concat(resourceId('Microsoft.Network/loadBalancers', variables('loadBalancerName')), '/backendAddressPools/', variables('loadBalancerBackendName'))]"
```
Comment (bmoore-msft) [inline 101-cassandra-existing-vnet/azuredeploy.json]: simplify - just use resourceId

File: 101-cassandra-existing-vnet/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+              },
+              "protocol": "Tcp",
+              "frontendPort": 9042,
+              "backendPort": 9042,
+              "enableFloatingIP": false,
+              "idleTimeoutInMinutes": 15,
+              "probe": {
+                "id": "[concat(resourceId('Microsoft.Network/loadBalancers', variables('loadBalancerName')), '/probes/', variables('loadBalancerProbeName'))]"
```
Comment (bmoore-msft) [inline 101-cassandra-existing-vnet/azuredeploy.json]: simplify - just use resourceId

File: 101-cassandra-existing-vnet/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+              "numberOfProbes": 2
+            }
+          }
+        ],
+        "inboundNatPools": [
+          {
+            "name": "[variables('loadBalancerNatPoolName')]",
+            "properties": {
+              "frontendIPConfiguration": {
+                "id": "[concat(resourceId('Microsoft.Network/loadBalancers', variables('loadBalancerName')), '/frontendIPConfigurations/', variables('loadBalancerFrontEndName'))]"
```
Comment (bmoore-msft) [inline 101-cassandra-existing-vnet/azuredeploy.json]: simplify - just use resourceId

File: 101-cassandra-existing-vnet/azuredeploy.parameters.json [hunk 11]
```diff
@@ -0,0 +1,33 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "clusterName": {
+      "value": "GEN-UNIQUE"
+    },
+    "adminUsername": {
+      "value": "azureuser"
```
Comment (bmoore-msft) [inline 101-cassandra-existing-vnet/azuredeploy.parameters.json]: GEN-UNIQUE

File: 101-cassandra-existing-vnet/azuredeploy.parameters.json [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+    },
+    "adminPassword": {
+      "value": "GEN-PASSWORD"
+    },
+    "nodeCount": {
+      "value": 2
```
Comment (bmoore-msft) [inline 101-cassandra-existing-vnet/azuredeploy.parameters.json]: remove the ones that have defaultValues defined in the template

File: 101-cassandra-existing-vnet/nested/cassandra-extension-vmss.json [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+            "type": "string"
+        }
+    },
+    "variables": {},
+    "resources": [
+        {
+            "type": "Microsoft.Compute/virtualMachineScaleSets",
+            "name": "[parameters('clusterName')]",
+            "location": "[resourceGroup().location]",
+            "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline 101-cassandra-existing-vnet/nested/cassandra-extension-vmss.json]: use a non-preview apiVersion

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**